include/Surelog/Common/FileSystem.h

```diff
@@ -0,0 +1,70 @@
+/*
+ Copyright 2022 chipsalliance
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+/*
+ * File:   FileSystem.cpp
+ * Author: hs
+ *
+ * Created on June 1, 2022, 3:00 AM
+ */
+
+#ifndef SURELOG_FILESYSTEM_H
+#define SURELOG_FILESYSTEM_H
+#pragma once
+
+#include <cstdint>
+#include <filesystem>
+#include <string_view>
+
+namespace SURELOG {
+class PathId;
+class SymbolId;
+class SymbolTable;
+
+class FileSystem {
+ public:
+  static FileSystem *getInstance();
+  static FileSystem *setInstance(FileSystem *const fileSystem);
+
+ public:
+  virtual PathId toPathId(const std::filesystem::path &path,
+                          SymbolTable *const symbolTable);
+  virtual const std::string &toSymbol(PathId id);
+  virtual std::filesystem::path toPath(PathId id);
+
+  virtual const std::filesystem::path &getCwd();
+  virtual PathId getCwd(SymbolTable *const symbolTable);
```

Comment:
Don't make the parameter `*const`, as it does not have any meaning but is just confusing
(it would have a meaning if the `const*` was left of the pointer).

Similar in other places of thie PR.

Reply:
Yes, but you typically would put that then in the _implementation_ part of the function, not the prototype. So in the API, you should keep it without the `*const`, in the implementation you then can choose to use it.

Reply:
`const*` tells the user that the input parameter is not changed. `*const` tells the function to not assign anything to the variable, but it doesn't make any difference to the caller.

Tthe 'const' definitely does not make a difference in the signature if it refers to the _variable_ being assigned.

The following signatures are identical:
```
  void foo(int x);
  void foo(const int x);
```

Sounds like your editor is overzealous, does simplified matching or wants to remind you of something that might be there in error.